server/src/main/java/org/eclipse/lsp/cobol/core/model/tree/ProgramNode.java

```diff
@@ -162,4 +172,8 @@ public void addVariableDefinition(VariableNode node) {
         .flatMap(Collection::stream)
         .collect(toList());
   }
+
+  private enum NodeState {
+    NEW, WAIT_FOR_VARIABLES, DONE;
+  }
```

Comment:
Shouldn't it either belong to the Node or be a separate class?

Reply:
Each node can have its own stages. I don't expect to have one set of states for all nodes. What if some node requires more than three states?

Reply:
Makes sense